utbot-framework/src/main/kotlin/org/utbot/framework/context/spring/SpringIntegrationTestJavaFuzzingContext.kt

```diff
@@ -37,28 +37,35 @@ class SpringIntegrationTestJavaFuzzingContext(
         private val logger = KotlinLogging.logger {}
     }
 
-    override val valueProvider: JavaValueProvider =
+    private val springBeanValueProvider: JavaValueProvider =
         SpringBeanValueProvider(
             idGenerator,
             beanNameProvider = { classId ->
                 springApplicationContext.getBeansAssignableTo(classId).map { it.beanName }
             },
             relevantRepositories = relevantRepositories
         )
+
+    override val valueProvider: JavaValueProvider =
+        springBeanValueProvider
+            .with(ModifyingWithMethodsProviderWrapper(classUnderTest, springBeanValueProvider))
             .withFallback(ValidEntityValueProvider(idGenerator, onlyAcceptWhenValidIsRequired = true))
             .withFallback(EmailValueProvider())
             .withFallback(NotBlankStringValueProvider())
             .withFallback(NotEmptyStringValueProvider())
             .withFallback(
                 delegateContext.valueProvider
-                    .except { p -> p is ObjectValueProvider }
-                    .with(anyObjectValueProvider(idGenerator, shouldMutateWithMethods = true))
-                    .with(ValidEntityValueProvider(idGenerator, onlyAcceptWhenValidIsRequired = false))
+                    .withProviderAndModifyingMethodsBuddy(anyObjectValueProvider(idGenerator))
+                    .withProviderAndModifyingMethodsBuddy(ValidEntityValueProvider(idGenerator, onlyAcceptWhenValidIsRequired = false))
                     .with(createGeneratedFieldValueProviders(relevantRepositories, idGenerator))
                     .withFallback(AnyDepthNullValueProvider)
             )
             .preserveProperties()
 
+    private fun JavaValueProvider.withProviderAndModifyingMethodsBuddy(provider: JavaValueProvider): JavaValueProvider =
+        this.with(provider)
+            .with(ModifyingWithMethodsProviderWrapper(classUnderTest, this))
```

Comment:
I suggest replace this function with the following and use it like this `baseProvider.with(otherProvider.withModifyingMethodsBuddy())`
```suggestion
    private fun JavaValueProvider.withModifyingMethodsBuddy(): JavaValueProvider =
        this.with(ModifyingWithMethodsProviderWrapper(classUnderTest, this))
```